modules/setting/setting.go

```diff
@@ -664,6 +665,12 @@ func NewContext() {
 	// This value is empty if site does not have sub-url.
 	AppSubURL = strings.TrimSuffix(url.Path, "/")
 	AppSubURLDepth = strings.Count(AppSubURL, "/")
+	// Check if Domain differs from AppURL domain than update it to AppURL's domain
+	// TODO: Can be replaced with url.Hostname() when minimal GoLang version is 1.8
+	urlHostname := strings.SplitN(url.Host, ":", 2)[0]
+	if urlHostname != Domain && net.ParseIP(urlHostname) == nil {
+		Domain = urlHostname
```

Comment:
Domain as it used should contain port at least for the use maid in : https://github.com/go-gitea/gitea/blob/2ef8b8bcbeeb7efabb6cfd247ad6227b0e2e21cd/modules/context/repo.go#L138 
`path.Join(setting.Domain, setting.AppSubURL, owner, repo)`

Reply:
It would be better to update the ComposeGoGetImport to add HTTP_PORT

Reply:
@sapk no it should not contain port as it generates `go` `import` part of repository and go imports can not contain port

Reply:
Sorry mislead HTTP_PORT is not necessary the same as the public port ... but we still need to add the url port to the go-import.

Reply:
ok sorry that kind of strange how it works if the host is not on defaut port ?

Reply:
oh sorry I understand that the repo url after package that contain the port.

Reply:
Yes git url is generated in other place. This is just import part